src/HospitalLibrary/Rooms/EquipmentReallocationService.cs

```diff
@@ -0,0 +1,212 @@
+            intervalsB = await GetAvailableIntervals(intervalsB, date, duration);
+
+            List<TimeInterval> intervals = await getShared(intervalsA, intervalsB, duration);
+            
+
+            return await smallerIntervals(intervals, duration);
+     
+        }
+
+        private async Task<List<TimeInterval>> smallerIntervals(List<TimeInterval> intervals, TimeSpan duration)
+        {
+            List<TimeInterval> smalIntervals = new List<TimeInterval>();
+
+            foreach (var interval in intervals)
+            {
+                smalIntervals.AddRange(snipInterval(interval, duration));
+            }
+            return smalIntervals;
+        }
+
+        
+        private List<TimeInterval> snipInterval(TimeInterval interval, TimeSpan duration)
+        {
+            List<TimeInterval> snippedInterval = new List<TimeInterval>();
+            var start=interval.Start;
+            int reps = interval.NumberOfTimespans(duration);
+            for (int i = 0; i < reps; i++)
+            {
+                snippedInterval.Add(new TimeInterval(start,start=start + duration));
+            }
+            return snippedInterval;
+        }
+
+        public async Task<List<TimeInterval>> GetTakenIntervals(int roomId, DateTime date)
+        {
+            List<TimeInterval> roomTimeIntervals =
+                await _unitOfWork.AppointmentRepository.GetAllRoomTakenIntervalsForDate(roomId, date);
+            List<TimeInterval> roomReallocationIntervals =
+                await _unitOfWork.EquipmentReallocationRepository.GetAllRoomTakenInrevalsForDate(roomId, date);
+
+
+            List<TimeInterval> mixedIntervals = roomReallocationIntervals.Concat(roomTimeIntervals).ToList();
+            return mixedIntervals = mixedIntervals.OrderBy(x => x.Start).ToList();
+        }
+
+        private async Task<List<TimeInterval>> GetAvailableIntervals(List<TimeInterval> takenIntervals, DateTime date, TimeSpan duration)
+        {
+            List<TimeInterval> available = new List<TimeInterval>();
+            DateTime b = date.Date.AddHours(8);
+
+            foreach (TimeInterval interval in takenIntervals)
+            {
+                if (interval.Start.Subtract(b) >= duration)
+                    available.Add(new TimeInterval(b, interval.Start));
+                b = interval.End;
+            }
+            date=date.Date.AddHours(21);
+            if (date.Subtract(b) >= duration)
+                available.Add(new TimeInterval(b,date));
+            return available;
+        }
+
+        private async Task<List<TimeInterval>> getShared(List<TimeInterval> intervalsA, List<TimeInterval> intervalsB ,TimeSpan duration)
+        {
+            List<TimeInterval> shared = new List<TimeInterval>(); 
+            foreach(var interval in intervalsA)
+            {
+                foreach(var interval2 in intervalsB)
+                {
+                    var a = AreCompatible(duration, interval, interval2);
```

Comment:
When method has a name that starts with Are, Is etc. it shoud return boolean. Here it's doing more work than the name says